crates/uv-cache/src/lib.rs

```diff
@@ -35,6 +36,24 @@ mod wheel;
 /// Must be kept in-sync with the version in [`CacheBucket::to_str`].
 pub const ARCHIVE_VERSION: u8 = 0;
 
+/// Error locking a cache entry or shard
+#[derive(Debug, Error)]
+pub enum LockCacheError {
+    #[error("Could not create path")]
+    CreateRoot(#[from] io::Error),
+    #[error("Could not acquire lock")]
+    Acquire(#[from] LockedFileError),
+}
+
+/// Error initialising the cache
+#[derive(Debug, Error)]
+pub enum InitCacheError {
```

Comment:
nit: We usually have one top level error enum per module or crate. It's a developer convenience that allows the functions and methods in the module or crate to all return the same error type, where it's easier to add new variants to a single type.

Reply:
This isn't a rule or anything fwiw, it's more a pattern that seems common, and there's definitely exceptions.